ci-operator/step-registry/ipi/conf/aws/proxy/ipi-conf-aws-proxy-commands.sh

```diff
@@ -0,0 +1,392 @@
+
+  ProxyInstanceProfile:
+    Type: "AWS::IAM::InstanceProfile"
+    Properties:
+      Path: "/"
+      Roles:
+      - Ref: "ProxyIamRole"
+
+  ProxySecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Cluster Proxy Security Group
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 22
+        ToPort: 22
+        CidrIp: 0.0.0.0/0
+      - IpProtocol: tcp
+        ToPort: 3128
+        FromPort: 3128
+        CidrIp: !Ref AllowedProxyCidr
+      - IpProtocol: tcp
+        ToPort: 19531
+        FromPort: 19531
+        CidrIp: !Ref AllowedProxyCidr
+      VpcId: !Ref VpcId
+
+  ProxyInstance:
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !Ref RhcosAmi
+      IamInstanceProfile: !Ref ProxyInstanceProfile
+      InstanceType: "i3.large"
+      NetworkInterfaces:
+      - AssociatePublicIpAddress: "true"
+        DeviceIndex: "0"
+        GroupSet:
+        - !Ref "ProxySecurityGroup"
+        SubnetId: !Ref "PublicSubnet"
+      UserData:
+        Fn::Base64: !Sub
+        - '{"ignition":{"config":{"replace":{"source":"\${IgnitionLocation}","verification":{}}},"timeouts":{},"version":"2.1.0"},"networkd":{},"passwd":{},"storage":{},"systemd":{}}'
+        - {
+          IgnitionLocation: !Ref ProxyIgnitionLocation
+        }
+
+Outputs:
+  ProxyPublicIp:
+    Description: The proxy node public IP address.
+    Value: !GetAtt ProxyInstance.PublicIp
+  ProxyId:
+    Description: The proxy node instanceId.
+    Value: !Ref ProxyInstance
+EOF
+}
+
+set -x
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+# TODO: move to image
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+EXPIRATION_DATE=$(date -d '4 hours' --iso=minutes --utc)
+TAGS="Key=expirationDate,Value=${EXPIRATION_DATE}"
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+
+PROXY_IMAGE=registry.svc.ci.openshift.org/origin/4.5:egress-http-proxy
+
+PROXY_NAME="$(/tmp/yq r ${CONFIG} 'metadata.name')"
+REGION="$(/tmp/yq r ${CONFIG} 'platform.aws.region')"
+
+RHCOS_AMI="$(jq -r .amis[\"${REGION}\"].hvm /var/lib/openshift-install/rhcos.json)"
+
+ssh_pub_key=$(<"${CLUSTER_PROFILE_DIR}/ssh-publickey")
+
+# get the VPC ID from a subnet -> subnet.VpcId
+aws_subnet="$(/tmp/yq r ${CONFIG} 'platform.aws.subnets[0]')"
+vpc_id="$(aws --region "${REGION}" ec2 describe-subnets --subnet-ids ${aws_subnet} | jq -r .[][0].VpcId)"
+
+# for each subnet:
+# aws ec2 describe-route-tables --filters Name=association.subnet-id,Values=${value} | grep '"GatewayId": "igw.*'
+  # if $? then use it as the public subnet
+
+public_subnet=""
+for subnet in $(/tmp/yq r -P ${CONFIG} 'platform.aws.subnets' | sed 's/- //g'); do
+  if aws --region "${REGION}" ec2 describe-route-tables --filters Name=association.subnet-id,Values="${subnet}" | grep '"GatewayId": "igw.*' 1>&2 > /dev/null; then
+    public_subnet="${subnet}"
+    break
+  fi
+done
+
```

Comment:
and a guard here for "echo "Could not find a public subnet in ${SUBNETS}" && exit 1`?